### PR TITLE
feat(icon-toggle): Annotate mdc-icon-toggle for closure

### LINKED ISSUE #337: Annotate mdc-icon-toggle for closure

### PR DESCRIPTION
**Pull request by nckh**
Closes #337 

Includes currently pending changes from `mdc-ripple/index.js` in #867

### CODE REVIEW

File: packages/mdc-icon-toggle/foundation.js [hunk 1]
```diff
@@ -84,22 +104,26 @@ export default class MDCIconToggleFoundation extends MDCFoundation {
...
   isOn() {
     return this.on_;
   }
 
+  /** @param {boolean=} isOn */
   toggle(isOn = !this.on_) {
     this.on_ = isOn;
 
     const {ARIA_LABEL, ARIA_PRESSED} = MDCIconToggleFoundation.strings;
     const {content, label, cssClass} = this.on_ ? this.toggleOnData_ : this.toggleOffData_;
```
Comment (lynnjepsen) [inline packages/mdc-icon-toggle/foundation.js]: is this line redundant to the one underneath?

File: packages/mdc-icon-toggle/foundation.js [hunk 2]
```diff
@@ -121,41 +145,74 @@ export default class MDCIconToggleFoundation extends MDCFoundation {
...
+
+
+/** @record */
+class IconToggleState {}
+
+/**
+ * @type {string|undefined}
+ * @export
+ */
+IconToggleState.prototype.label;
```
Comment (lynnjepsen) [inline packages/mdc-icon-toggle/foundation.js]: can you add a comment for what label, content, and cssClass is? And in what situation they can be undefined?

File: packages/mdc-ripple/index.js [hunk 3]
```diff
@@ -114,3 +114,25 @@ export class MDCRipple extends MDCComponent {
     this.unbounded = 'mdcRippleIsUnbounded' in this.root_.dataset;
   }
 }
+
+/**
+ * See Material Design spec for more details on when to use ripples.
+ * https://material.io/guidelines/motion/choreography.html#choreography-creation
+ * @record
+ */
+class RippleCapableSurface {}
```
Comment (lynnjepsen) [inline packages/mdc-ripple/index.js]: wait for that other PR to get merged, then you'll have to sync this one up.

**Comment by lynnjepsen** [comment]
I've copied this to https://github.com/material-components/material-components-web/pull/958, along with my requested changes. We generally close PRs that are non-responsive, but I didn't want to lose this valuable work before closing out the PR.

**Comment by nckh** [comment]
Sorry Lynn!! Got busy plus a few holidays which did not make me super responsive recently. 
I'll make sure to respond within a week from now on.

**Pull request closed without merging**